frontend/src/components/pages/your_clubs/YourClubs.tsx

```diff
@@ -1,10 +1,287 @@
+        <DisplayListedClubs clubsToDisplay={listedClubs} />
+      </div>
+    </div>
+  );
+}
+
+interface DisplayListedClubsProps {
+  clubsToDisplay: ClubProps[],
+}
+
+/**
+ * Displays up to the number of clubs that the user has requested to to
+ * the page.
+ */
+function DisplayListedClubs(props: DisplayListedClubsProps) {
+  const classes = useStyles();
+  const clubsToDisplay = props.clubsToDisplay;
+
+  /** If clubs to display is not yet defined, or 'None' are chosen to display. */
+  if (clubsToDisplay ? /** defined */ true : /** undefined */ false)  {
+    return (
+      <div className={classes.listedClubsContainer}>
+        {clubsToDisplay.map((item, index) => (
+          <div className={classes.club} key={item.name}>
+            <Box border={1} borderColor="text.primary" borderRadius={16}>
+              <div className={classes.clubContent}>
+                <h2 className={classes.clubTitle}>{item.name}</h2>
+                <LibraryBooksRoundedIcon className={classes.clubPhoto} />
+                <LoadClubDescription description={item.description} />
+                <LoadBookInfo book={item.currentBook} />
+                <LoadContentWarnings contentWarnings={item.contentWarnings} />
+                <div className={classes.break}></div>
+                <LoadButtons />
+              </div>
+            </Box>
+          </div>
+        ))}
+      </div>
+    );
+  } else {
+    return (<div></div>);
+  }
+}
+
+interface LoadClubDescriptionProps {
+  description: string;
+};
+
+/** Displays the club's description. */
+function LoadClubDescription(props: LoadClubDescriptionProps) {
+  const classes = useStyles();
+
+  return (
+    <div>
+      <div className={classes.break}></div>
+      <p className={classes.textElement}>
+        <b>Description: </b><br/>
+        {props.description}
+      </p>
+    </div>
+  );
+}
+
+interface LoadBookInfoProps {
+  book: BookProps;
+};
+
+/** Displays the title and author of the club's current book. */
+function LoadBookInfo(props: LoadBookInfoProps) {
+  const classes = useStyles();
+
+  return (
+    <div>
+      <p className={classes.textElement} style={{marginRight: 20}}>
+        <b>Current Book: </b><br/>
+        {' ' + props.book.title}<br/>
+        by {props.book.author}
+      </p>
+    </div>
+  );
+}
+
+/** Displays content warnings associated with the club's current book. */
+function LoadContentWarnings(props) {
+  const classes = useStyles();
+  const contentWarnings = props.contentWarnings;
+
   return (
     <div>
-      <h2>Your Clubs</h2>
-      <p>This is the Your Clubs page.</p>
+      <p  className={classes.textElement}><b>Content Warnings</b></p>
+      {contentWarnings.map((item, index) => (
+        <div>
+          - <b>{item}<br/></b>
+        </div>
+      ))}
+    </div>
+  );
+}
+
+/**
+ * Displays two buttons, one for viewing the club and another for leaving
+ * the club.
+ */
+function LoadButtons () {
```

Comment:
nit: `Buttons` is too generic of a name. This could probably be part of the DisplayListedClubs component since I don't forsee these specific buttons having a use outside of the app. Otherwise, something like `ClubActionButtons` would be cleaner